### PR TITLE
Added segment tree in java

### LINKED ISSUE #518: Creating segment tree in java
I will implement a segment tree in java

### PR DESCRIPTION
**Pull request by MadKat13**
This pull request fixes #518

### CODE REVIEW

File: Segment_trees/SegmentTree.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+public class SegmentTree {
```
Comment (t2013anurag) [inline Segment_trees/SegmentTree.java]: move the file inside Trees/segment_trees

File: Segment_trees/SegmentTree.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            tree[nodeposition] = arr[segstart];
+            return arr[segstart];
+        }
+
+        int mid = segstart + (segend-segstart)/2;
+        tree[nodeposition] = buildTree(arr,segstart,mid,nodeposition*2+1) +
+                buildTree(arr,mid+1,segend, nodeposition*2+2);
+        return tree[nodeposition];
+    }
+
```
Comment (t2013anurag) [inline Segment_trees/SegmentTree.java]: remove extra new line.

File: Segment_trees/SegmentTree.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        System.out.println("Sum of values in given range = " +
+              tree.getSum(1, 3));
+
+        tree.updateValue(arr, 1, 10);
+
+        System.out.println("Updated sum of values in given range = " +
+            tree.getSum(1, 3));
+    }
+
```
Comment (t2013anurag) [inline Segment_trees/SegmentTree.java]: remove this line.

**Comment by MadKat13** [comment]
Made the changes according to you @t2013anurag

**Pull request merged**